### DISCUSSION TITLE
GET Chat History / Prompt Completion Pairs

### DISCUSSION THREAD

**Discussion by anmolGup123** [FlowiseAI/Flowise#1181] [Q&A]
Does flowise have a chat history API? I know there's an API where you can POST a history array but how do I get chat history  or Prompt completion pairs of flow? I see Langfuse fetched each prompt completion pair, so there must be some API , I just can't find it! I tried using https://api.flowise.com/api/v1/prompt_completion_pairs and https://api.flowise.com/api/v1/chathistory but getting 401

**Comment by HenryHengZJ** [comment]
here - `/api/v1/chatmessage/{yourchatflowid}`, and you can query by `?sessionId={yoursessionid}`. If you have FW instance authenticated with username and password, you have to pass in those as Basic Auth - https://stackoverflow.com/a/51500400

**Comment by anmolGup123** [comment]
Thank you so much!

Cheers,
Anmol
